cardano-api/src/Cardano/Api/ProtocolParameters.hs

```diff
@@ -554,206 +714,75 @@ data GenesisParameters =
+        return pp
+
+toProtocolParamsShelley :: Shelley.PParams ledgerera -> ProtocolParameters era
+toProtocolParamsShelley pparams =
+   ProtocolParameters
+        { protocolParamProtocolVersion     = (\(Shelley.ProtVer a b) -> (a,b))
+                                               $ Shelley._protocolVersion pparams
+        , protocolParamDecentralization    = Shelley.unitIntervalToRational $ Shelley._d pparams
+        , protocolParamExtraPraosEntropy   = fromPraosNonce $ Shelley._extraEntropy pparams
+        , protocolParamMaxBlockHeaderSize  = Shelley._maxBHSize pparams
+        , protocolParamMaxBlockBodySize    = Shelley._maxBBSize pparams
+        , protocolParamMaxTxSize           = Shelley._maxTxSize pparams
+        , protocolParamTxFeeFixed          = Shelley._minfeeB pparams
+        , protocolParamTxFeePerByte        = Shelley._minfeeA pparams
+        , protocolParamMinUTxOValue        = Just . fromShelleyLovelace $ Shelley._minUTxOValue pparams
+        , protocolParamStakeAddressDeposit = fromShelleyLovelace $ Shelley._keyDeposit pparams
+        , protocolParamStakePoolDeposit    = fromShelleyLovelace $ Shelley._poolDeposit pparams
+        , protocolParamMinPoolCost         = fromShelleyLovelace $ Shelley._minPoolCost pparams
+        , protocolParamPoolRetireMaxEpoch  = Shelley._eMax pparams
+        , protocolParamStakePoolTargetNum  = Shelley._nOpt pparams
+        , protocolParamPoolPledgeInfluence = Shelley._a0 pparams
+        , protocolParamMonetaryExpansion   = Shelley.unitIntervalToRational $ Shelley._rho pparams
+        , protocolParamTreasuryCut         = Shelley.unitIntervalToRational $ Shelley._tau pparams
+        , protocolParamUTxOCostPerByte     = Nothing
+        , protocolParamCostModels          = mempty
+        , protocolParamPrices              = Nothing
+        , protocolParamMaxTxExUnits        = Nothing
+        , protocolParamMaxBlockExUnits     = Nothing
+        , protocolParamMaxValSize          = Nothing
+        }
+
+
+checkProtocolVersion
+  :: ShelleyBasedEra era
+  -> (Natural, Natural)
+  -> Either ProtocolParametersError ()
+checkProtocolVersion sbe protoVer  =
+  let expectedPVer = case sbe of
+                       ShelleyBasedEraShelley -> (2,0)
+                       ShelleyBasedEraAllegra -> (3,0)
+                       ShelleyBasedEraMary -> (4,0)
+              --      ShelleyBasedEraAlonzo -> (5,0)
+  in checkCondition (protoVer == expectedPVer)
+       $ ProtocolParametersErrorWrongVersion expectedPVer protoVer
+
+checkCondition :: Bool -> ProtocolParametersError -> Either ProtocolParametersError ()
+checkCondition True _ = Right ()
+checkCondition False ppe = Left ppe
 
-
-fromShelleyGenesis :: Shelley.ShelleyGenesis era -> GenesisParameters
 fromShelleyGenesis
+  :: ShelleyBasedEra era
+  -> Shelley.ShelleyGenesis (ShelleyLedgerEra era)
+  -> Either ProtocolParametersError (GenesisParameters era)
+fromShelleyGenesis
```

Comment:
I don't understand the changes to this function. It doesn't cover Alonzo yet but we hypothesise that we will need to split based on era? But if so we already know we can share the impl for the first three eras, so we don't need to duplicate those. Worst case we can define it once in a where clause and call it three times for the three existing era cases, plus something new for the Alonzo era.

If we don't yet know if we need to change it, we can postpone the change until we know.